exchanges/gateio/gateio_websocket.go

```diff
@@ -431,41 +416,31 @@ func (g *Gateio) processOrderbookSnapshot(data []byte) error {
 		Exchange:        g.Name,
 		Pair:            pair,
 		Asset:           asset.Spot,
-		LastUpdated:     snapshot.UpdateTimeMs.Time(),
-		LastUpdateID:    snapshot.LastUpdateID,
+		LastUpdated:     data.UpdateTimeMs.Time(),
+		LastUpdateID:    data.LastUpdateID,
 		VerifyOrderbook: g.CanVerifyOrderbook,
 	}
-	bases.Bids = make([]orderbook.Item, len(snapshot.Bids))
-	bases.Asks = make([]orderbook.Item, len(snapshot.Asks))
-	var price float64
-	var amount float64
+	bases.Asks = make([]orderbook.Item, len(data.Asks))
 	for x := range bases.Asks {
-		price, err = strconv.ParseFloat(snapshot.Asks[x][0], 64)
+		bases.Asks[x].Price, err = strconv.ParseFloat(data.Asks[x][0], 64)
 		if err != nil {
 			return err
 		}
-		amount, err = strconv.ParseFloat(snapshot.Asks[x][1], 64)
+		bases.Asks[x].Amount, err = strconv.ParseFloat(data.Asks[x][1], 64)
 		if err != nil {
 			return err
 		}
-		bases.Asks[x] = orderbook.Item{
-			Amount: amount,
-			Price:  price,
-		}
 	}
+	bases.Bids = make([]orderbook.Item, len(data.Bids))
 	for x := range bases.Bids {
```

Comment:
```suggestion
	for x := range data.Bids {
```

Even though all these will technically work due to `make` and ranging over it, better to be precise 😄